cmake/ConfigureCompiler.cmake

```diff
@@ -10,6 +10,7 @@ env_set(USE_GCOV OFF BOOL "Compile with gcov instrumentation")
 env_set(USE_MSAN OFF BOOL "Compile with memory sanitizer. To avoid false positives you need to dynamically link to a msan-instrumented libc++ and libc++abi, which you must compile separately. See https://github.com/google/sanitizers/wiki/MemorySanitizerLibcxxHowTo#instrumented-libc.")
 env_set(USE_TSAN OFF BOOL "Compile with thread sanitizer. It is recommended to dynamically link to a tsan-instrumented libc++ and libc++abi, which you can compile separately.")
 env_set(USE_UBSAN OFF BOOL "Compile with undefined behavior sanitizer")
+env_set(USE_INSTRUMENTATION OFF BOOL "Compile with instrumentation, it's primary used for Profile Guided Optimization, it can be used for code coverage too")
```

Comment:
This statement is little confusing because we have another parameter for gcov. Also, `USE_INSTRUMENTATION` doesn't sound very intuitive. Maybe something like `PGO_MODE={GENERATE|USE}`? We will need to add another branch for "use", correct?

Reply:
we will need to add another string variable for "use".
Acorrding to https://clang.llvm.org/docs/UsersManual.htm,   "-fprofile-instr-generate : Generate instrumented code to collect execution counts into default.profraw file",  " -fprofile-instr-use: Use instrumentation data for profile-guided optimization". For the later, shall we call them generate_instr_profile and use_instr_profile ?

Reply:
I'm just suggesting to change the CMake parameter name.
`USE_INSTRUMENTATION` is little ambiguous, I think. We need a parameter that can imply "Set this flag to build FDB as an instrumentation build to generate profiles".
I think it's easy to use the same CMake parameter to switch `generate` and `use`, but having 2 different parameters is OK too.
e.g.
`GENERATE_INSTR_PROFILE` and `USE_INSTR_PROFILE`
as you suggested.
In fact, this can be a better option, as you can then pass the profile file to `USE_INSTR_PROFILE=<prof>`

Reply:
Please take another look